src/test/java/de/tum/in/www1/artemis/AchievementIntegrationTest.java

```diff
@@ -0,0 +1,160 @@
+package de.tum.in.www1.artemis;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.test.context.support.WithMockUser;
+
+import de.tum.in.www1.artemis.domain.Achievement;
+import de.tum.in.www1.artemis.domain.Course;
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.domain.enumeration.Rank;
+import de.tum.in.www1.artemis.repository.AchievementRepository;
+import de.tum.in.www1.artemis.repository.CourseRepository;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.service.AchievementService;
+import de.tum.in.www1.artemis.util.DatabaseUtilService;
+import de.tum.in.www1.artemis.util.RequestUtilService;
+
+public class AchievementIntegrationTest extends AbstractSpringIntegrationBambooBitbucketJiraTest {
+
+    @Autowired
+    DatabaseUtilService database;
+
+    @Autowired
+    RequestUtilService request;
+
+    @Autowired
+    AchievementService achievementService;
+
+    @Autowired
+    AchievementRepository achievementRepository;
+
+    @Autowired
+    UserRepository userRepository;
+
+    @Autowired
+    CourseRepository courseRepository;
+
+    private User student;
+
+    private User instructor;
+
+    private Course course;
+
+    private Achievement achievement;
+
+    @BeforeEach
+    public void initTestCase() {
+        var users = database.addUsers(1, 0, 1);
+        student = users.get(0);
+        instructor = users.get(1);
+        instructor.setGroups(new HashSet<>(Arrays.asList("instructor")));
+        course = database.addEmptyCourse();
+
+        achievement = new Achievement();
+        achievement.setId(1L);
```

Comment:
When you save new entities to the database, they should **not** contain an id already!

Reply:
removed